File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/backend/php/fenix/src/controller/BancoController.php
<?php

require './src/service/BancoService.php';
require './src/model/transiente/RetornoJsonErro.php';
require './src/model/transiente/Filtro.php';

class BancoController
{

    public function consultarLista($request, $response, $args)
    {
        try {
            if (count($request->getQueryParams()) > 0) {
                $filtro = new Filtro($request->getQueryParams()['filter']);
                $listaConsulta = BancoService::consultarListaFiltroValor($filtro->campo, $filtro->valor);
            } else {
                $listaConsulta = BancoService::consultarLista();
            }
            $retorno = json_encode($listaConsulta);
            $response->getBody()->write($retorno);
            return $response->withHeader('Content-Type', 'application/json');
        } catch (Exception $e) {
            $jsonErro = new RetornoJsonErro(500, 'Erro no Servidor [Consultar Lista Banco]', $e);
            $retorno = json_encode($jsonErro);
            $response->getBody()->write($retorno);
            return $response
                ->withStatus(500)
                ->withHeader('Content-Type', 'application/json');
        }
    }
    
    // public function consultarListaFiltroValor($request, $response, $args)
    // {
    //     try {
    //         $listaConsulta = BancoService::consultarListaFiltroValor($args['campo'], $args['valor']);
    //         $retorno = json_encode($listaConsulta);
    //         $response->getBody()->write($retorno);
    //         return $response->withHeader('Content-Type', 'application/json');
    //     } catch (Exception $e) {
    //         $jsonErro = new RetornoJsonErro(500, 'Erro no Servidor [Consultar Lista Filtro Valor Banco]', $e);
    //         $retorno = json_encode($jsonErro);
    //         $response->getBody()->write($retorno);
    //         return $response
    //             ->withStatus(500)
    //             ->withHeader('Content-Type', 'application/json');
    //     }
    // }

    public function consultarObjeto($request, $response, $args)
    {
        try {
            $objeto = BancoService::consultarObjeto($args['id']);

            if ($objeto == null) {
                $jsonErro = new RetornoJsonErro(404, 'Registro não localizado [Consultar Objeto Banco]', null);
                $retorno = json_encode($jsonErro);
                $response->getBody()->write($retorno);
                return $response
                    ->withStatus(404)
                    ->withHeader('Content-Type', 'application/json');
            } else {
                $retorno = json_encode($objeto);
                $response->getBody()->write($retorno);
                return $response
                    ->withHeader('Content-Type', 'application/json');
            }
        } catch (Exception $e) {
            $jsonErro = new RetornoJsonErro(500, 'Erro no Servidor [Consultar Objeto Banco]', $e);
            $retorno = json_encode($jsonErro);
            $response->getBody()->write($retorno);
            return $response
                ->withStatus(500)
                ->withHeader('Content-Type', 'application/json');
        }
    }

    public function inserir($request, $response, $args)
    {
        try {
            // pegar o objeto da requisição
            $bancoJson = json_decode($request->getBody());
            $banco = new Banco($bancoJson);

            BancoService::salvar($banco);

            $retorno = json_encode($bancoJson);
            $response->getBody()->write($retorno);

            return $response
                ->withStatus(201)
                ->withHeader('Content-Type', 'application/json');
        } catch (Exception $e) {
            $jsonErro = new RetornoJsonErro(500, 'Erro no Servidor [Inserir Banco]', $e);
            $retorno = json_encode($jsonErro);
            $response->getBody()->write($retorno);
            return $response
                ->withStatus(500)
                ->withHeader('Content-Type', 'application/json');
        }
    }

    public function alterar($request, $response, $args)
    {
        // pegar o objeto da requisição
        $bancoJson = json_decode($request->getBody());

        // valida objeto
        $banco = new Banco($bancoJson);
        $mensagemErro = $banco->validarObjetoRequisicao($bancoJson, $args['id']);
        if ($mensagemErro != '') {
            $jsonErro = new RetornoJsonErro(400, $mensagemErro, null);
            $retorno = json_encode($jsonErro);
            $response->getBody()->write($retorno);
            return $response
                ->withStatus(400)
                ->withHeader('Content-Type', 'application/json');
        }

        try {
            $dbBanco = BancoService::consultarObjeto($bancoJson->id);

            if ($dbBanco == null) {
                $jsonErro = new RetornoJsonErro(400, 'Objeto inválido [Alterar Banco]', null);
                $retorno = json_encode($jsonErro);
                $response->getBody()->write($retorno);
                return $response
                    ->withStatus(400)
                    ->withHeader('Content-Type', 'application/json');
            } else {
                $dbBanco->mapear($banco);
                BancoService::salvar($dbBanco);

                $retorno = json_encode($bancoJson);
                $response->getBody()->write($retorno);
                return $response
                    ->withHeader('Content-Type', 'application/json');
            }
        } catch (Exception $e) {
            $jsonErro = new RetornoJsonErro(500, 'Erro no Servidor [Alterar Banco]', $e);
            $retorno = json_encode($jsonErro);
            $response->getBody()->write($retorno);
            return $response
                ->withStatus(500)
                ->withHeader('Content-Type', 'application/json');
        }
    }

    public function excluir($request, $response, $args)
    {
        try {
            $dbBanco = BancoService::consultarObjeto($args['id']);

            if ($dbBanco == null) {
                $jsonErro = new RetornoJsonErro(400, 'Objeto inválido [Excluir Banco]', null);
                $retorno = json_encode($jsonErro);
                $response->getBody()->write($retorno);
                return $response
                    ->withStatus(400)
                    ->withHeader('Content-Type', 'application/json');
            } else {
                BancoService::excluir($dbBanco);

                return $response
                    ->withStatus(200)
                    ->withHeader('Content-Type', 'application/json');
            }
        } catch (Exception $e) {
            $jsonErro = new RetornoJsonErro(500, 'Erro no Servidor [Excluir Banco]', $e);
            $retorno = json_encode($jsonErro);
            $response->getBody()->write($retorno);
            return $response
                ->withStatus(500)
                ->withHeader('Content-Type', 'application/json');
        }
    }
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/backend/c-sharp/T2TiERPFenix/T2TiERPFenix/Controllers/BancoController.cs
using Microsoft.AspNetCore.Mvc;
using System;
using T2TiERPFenix.Models;
using T2TiERPFenix.Repository;
using Microsoft.AspNetCore.Http;
using System.Collections.Generic;

namespace T2TiERPFenix.Controllers
{
    [Route("banco")]
    [Produces("application/json")]
    public class BancoController : Controller
    {
        private IRepositoryWrapper _repository;

        public BancoController(IRepositoryWrapper repository)
        {
            _repository = repository;
        }

        [HttpGet]
        public IActionResult ConsultarListaBanco([FromQuery]string filter)
        {
            try
            {
                IEnumerable<Banco> listaBanco;
                if (filter == null)
                {
                    listaBanco = _repository.Banco.ConsultarLista();
                } else
                {
                    // define o filtro
                    Filtro filtro = new Filtro(filter);
                    listaBanco = _repository.Banco.ConsultarListaFiltro(filtro);
                }
                return Ok(listaBanco);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Erro no Servidor [Consultar Lista Banco]", ex));
            }
        }

        //[HttpGet("{campo}/{valor}")]
        //public IActionResult ConsultarListaBancoFiltroValor(String campo, String valor)
        //{
        //    try
        //    {
        //        // define o filtro
        //        Filtro filtro = new Filtro();
        //        filtro.Campo = campo;
        //        filtro.Valor = valor;

        //        var listaBanco = _repository.Banco.ConsultarListaFiltro(filtro);

        //        return Ok(listaBanco);
        //    }
        //    catch (Exception ex)
        //    {
        //        return StatusCode(500, new RetornoJsonErro(500, "Erro no Servidor [Lista Banco Filtro Padrão]", ex));
        //    }
        //}

        [HttpGet("{id}", Name = "ConsultarObjetoBanco")]
        public IActionResult ConsultarObjetoBanco(int id)
        {
            try
            {
                var banco = _repository.Banco.ConsultarObjeto(id);

                if (banco == null)
                {
                    return StatusCode(404, new RetornoJsonErro(404, "Registro não localizado [Consultar Objeto Banco]", null));
                }
                else
                {
                    return Ok(banco);
                }
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Erro no Servidor [Consultar Objeto Banco]", ex));
            }
        }

        [HttpPost]
        public IActionResult InserirBanco([FromBody]Banco banco)
        {
            try
            {
                if (!ModelState.IsValid)
                {
                    return StatusCode(400, new RetornoJsonErro(400, "Objeto inválido [Inserir Banco]", null));
                }

                _repository.Banco.Inserir(banco);

                return CreatedAtRoute("ConsultarObjetoBanco", new { id = banco.Id }, banco);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Erro no Servidor [Inserir Banco]", ex));
            }
        }

        [HttpPut("{id}")]
        public IActionResult AlterarBanco([FromBody]Banco banco, int id)
        {
            try
            {
                if (!ModelState.IsValid)
                {
                    return StatusCode(400, new RetornoJsonErro(400, "Objeto inválido [Alterar Banco]", null));
                }

                if (banco.Id != id)
                {
                    return StatusCode(400, new RetornoJsonErro(400, "Objeto inválido [Alterar Banco] - ID do objeto difere do ID da URL.", null));
                }

                var dbBanco = _repository.Banco.ConsultarObjeto(banco.Id);

                if (dbBanco == null)
                {
                    return StatusCode(400, new RetornoJsonErro(400, "Objeto com ID inválido [Alterar Banco]", null));
                }

                _repository.Banco.Alterar(dbBanco, banco);

                return Ok(banco);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Erro no Servidor [Alterar Banco]", ex));
            }
        }

        [HttpDelete("{id}")]
        public IActionResult ExcluirBanco(int id)
        {
            try
            {
                var banco = _repository.Banco.ConsultarObjeto(id);

                _repository.Banco.Excluir(banco);

                return Ok();
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Erro no Servidor [Excluir Banco]", ex));
            }
        }

    }
}